### PR TITLE
feat: change onboarding acknowledgement flow

### PR DESCRIPTION
**Pull request by rosepuppy**
OLD:

https://github.com/user-attachments/assets/66e3a48a-9bba-4e73-9e43-76fa765311c5

NEW:

https://github.com/user-attachments/assets/360e9ab0-abbe-4028-bd6f-100752df1071

Update onboarding acknowledgement flow based on this https://linear.app/dydx/issue/OTE-318/update-onboarding-flow-to-reduce-friction-for-users

### CODE REVIEW

File: src/constants/account.ts [hunk 1]
```diff
@@ -3,7 +3,6 @@ import type { DydxAddress, EvmAddress } from './wallets';
 export enum OnboardingSteps {
   ChooseWallet = 'ChooseWallet',
   KeyDerivation = 'KeyDerivation',
-  AcknowledgeTerms = 'AcknowledgeTerms',
```
Comment (rosepuppy) [inline src/constants/account.ts:6]: No longer need AcknowledgeTerms as an OnboardingStep

**Comment by yogurtandjam** [comment]
General thought: why is the acknowledge link so small/in the corner? Can we make the acknowledge terms and deposit button open up that modal instead?

**Comment by rosepuppy** [comment]
> General thought: why is the acknowledge link so small/in the corner? Can we make the acknowledge terms and deposit button open up that modal instead?

I think the idea, is that we don't really need to show the modal unless the user wants to read the whole thing. The acknowledgement will be implicit when they deposit.

I think we were seeing drop offs during the ack modal and we want to skip that step.

File: src/views/forms/AccountManagementForms/DepositForm.tsx [hunk 2]
```diff
@@ -273,6 +287,10 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
 
         setIsLoading(true);
 
+        if (!hasAcknowledgedTerms) {
+          saveHasAcknowledgedTerms(true);
```
Comment (rosepuppy) [inline src/views/forms/AccountManagementForms/DepositForm.tsx:291]: Set acknowledgement on deposit

**Comment by yogurtandjam** [comment]
> > General thought: why is the acknowledge link so small/in the corner? Can we make the acknowledge terms and deposit button open up that modal instead?
> 
> I think the idea, is that we don't really need to show the modal unless the user wants to read the whole thing. The acknowledgement will be implicit when they deposit. Basically if they just click the deposit CTA, they don't need to view the ack modal at all.
> 
> I think we were seeing drop offs during the ack modal and we want to skip that step.

oh i see, i misinterpreted and thought the reason why the "acknowledge and deposit" button was greyed out on your screen was becasue you hadn't accepted terms and conditions, but it was just because no amount was populated for the deposit. is that correct?

**Comment by rosepuppy** [comment]
> oh i see, i misinterpreted and thought the reason why the "acknowledge and deposit" button was greyed out on your screen was becasue you hadn't accepted terms and conditions, but it was just because no amount was populated for the deposit. is that correct?

Correct

**Review by yogurtandjam — APPROVED** [review]

**Pull request merged**